import/index_java/src/org/vufind/index/MultiFormatCalculator.java

```diff
@@ -0,0 +1,448 @@
+            case 'E':
+            case 'F':
+                return "Map";
+            case 'G':
+                // We're going to rely on the 007 instead for Projected Media
+                //return "Slide";
+                return "";
+            case 'I':
+                return "SoundRecording";
+            case 'J':
+                return "MusicRecording";
+            case 'K':
+                return "Photo";
+            case 'O':
+            case 'P':
+                return "Kit";
+            case 'R':
+                return "PhysicalObject";
+            case 'T':
+                if (!isThesis(record)) {
+                    return "Manuscript";
+                }
+                break;
+        }
+        return "";
+    }
+
+    protected char getTypeOfComputerFile(ControlField fixedField) {
+        // Check the 008 for the type of computer file:
+        try {
+            return fixedField.getData().toUpperCase().charAt(26);
+        } catch (java.lang.StringIndexOutOfBoundsException e) {
+            // ignore errors (leave the string blank if out of bounds)
+            return ' ';
+        }
+    }
+
+    /**
+     * Determine whether a record is a conference proceeding.
+     *
+     * @param Record record
+     * @return boolean
+     */
+    protected boolean isConferenceProceeding(Record record) {
+        // Is there a main entry meeting name?
+        DataField conference1 = (DataField) record.getVariableField("111");
+        // The 711 could possibly have more then one entry, although probably unlikely
+        DataField conference2 = (DataField) null;
+        List conferenceFields = record.getVariableFields("711");
+        Iterator conferenceFieldsIter = conferenceFields.iterator();
+        if (conferenceFields != null) {
+            if (conferenceFieldsIter.hasNext()) {
+                conference2 = (DataField) conferenceFieldsIter.next();
+            }
+        }
+        return (conference1 != null || conference2 != null);
+    }
+
+    /**
+     * Determine whether a record is electronic in format.
+     *
+     * @param Record record
+     * @return boolean
+     */
+    protected boolean isElectronic(Record record) {
+        /* Example from Villanova of how to use holdings locations to detect online status;
+         * You can override this method in a subclass if you wish to use this approach.
+        DataField holdingsField = (DataField) record.getVariableField("852");
+        if (holdingsField != null) {
+            if (holdingsField.getSubfield('b') != null) {
+                String holdingsLocation = holdingsField.getSubfield('b').getData().toLowerCase();
+                if (holdingsLocation.equals("www") || holdingsLocation.equals("e-ref")) {
+                    return true;
+                }
+            }
+        }
+         */
+        DataField title = (DataField) record.getVariableField("245");
+        if (title != null) {
+            if (title.getSubfield('h') != null){
```

Comment:
Missing space :)

Reply:
There were actually two of these. Both fixed now. Thanks! :-)